python/cuda_parallel/tests/test_reduce_api.py

```diff
@@ -29,12 +28,12 @@ def op(a, b):
     temp_storage_size = reduce_into(None, d_input, d_output, h_init)
 
     # Allocate temporary storage
-    d_temp_storage = cuda.device_array(temp_storage_size, dtype=numpy.uint8)
+    d_temp_storage = cp.empty(temp_storage_size, dtype=np.uint8)
 
     # Run reduction
     reduce_into(d_temp_storage, d_input, d_output, h_init)
 
+    # Check the result is correct
     expected_output = 0
+    assert (d_output == expected_output).all()
     # example-end reduce-min
```

Comment:
suggestion: we keep test-specific code out of documentation:
```suggestion
    # example-end reduce-min
    assert (d_output == expected_output).all()
```

Reply:
Right now the example just shows `expected_output = 0` which is confusing. Adding this line serves two purposes:
1. Actually make comparison between expected and computed outputs
2. Show best practice when checking arrays residing on GPUs

For this case it might be an overkill since it's just a scalar, but by the time we move to more complex APIs it'd be helpful, and I prefer to set up a good precedence here.